clean/post_ohr.py

```diff
@@ -0,0 +1,537 @@
+        .str.replace(
+            "probation & parcole - adult",
+            "adult probation",
+            regex=True,
+        )
+        .str.extract(
+            r"^(\w{1}? ?\w+ ?\w+? ?\w+) ?(deceased|full-time|reserve|retired|part-time)? ? "
+            r"?(\w{1,2}\/\w{1,2}\/\w{4}) ? ?(\w{1,2}\/\w{1,2}\/\w{4})? ?(termination|(\w+)? "
+            r"?(resignation)|(\w+)? ?(resigned)|retired)?"
+        )
+    )
+    df.loc[:, "agency_7"] = agency7[0]
+    df.loc[:, "employment_status_7"] = agency7[1]
+    df.loc[:, "hire_date_7"] = agency7[2]
+    df.loc[:, "left_date_7"] = agency7[3]
+    df.loc[:, "left_reason_7"] = agency7[4]
+
+    agency8 = (
+        df.agency_8.astype(str)
+        .str.lower()
+        .str.strip()
+        .str.replace(r"(\w+) = (\w+)", r"\1 \2", regex=True)
+        .str.replace(
+            r"^orleans parish coroner\'s office", "orleans coroners office", regex=True
+        )
+        .str.replace(r"(\w+) Ã¢â‚¬â€ (\w+)", r"\1 \2", regex=True)
+        .str.replace(r"1st parish court", "first court", regex=False)
+        .str.replace(r"(\w+) Ã‚Â© (\w+)\/(\w+)\/(\w+)", r"\1 \2/\3/\4", regex=True)
+        .str.replace(r" Â© ", "", regex=True)
+        .str.replace(r"--BULL-TIME", "FULL-TIME", regex=True)
+        .str.replace(r"_(\w+)\/(\w+)\/(\w+)â€”_", r"\1/\2/\3", regex=True)
+        .str.replace(r" â€”= ", "", regex=True)
+        .str.replace(r" +", " ", regex=True)
+        .str.replace(r" & ", "", regex=True)
+        .str.replace(r" - ", "", regex=True)
+        .str.replace(r" _?â€\” ", "", regex=True)
+        .str.replace(r" _ ", "", regex=True)
+        .str.replace(r" = ", "", regex=True)
+        .str.replace(r"^ (\w+)", r"\1", regex=True)
+        .str.replace(r"^st (\w+)", r"st\1", regex=True)
+        .str.replace(r" of ", "", regex=True)
+        .str.replace(r" p\.d\.", " pd", regex=True)
+        .str.replace(r" pari?s?h? so", " so", regex=True)
+        .str.replace(r" Â§ ", "", regex=True)
+        .str.replace(r" â€˜", "", regex=True)
+        .str.replace(r"(\.|\,)", "", regex=True)
+        .str.replace(r"miss river", "river", regex=False)
+        .str.replace(r" ~ ", "", regex=False)
+        .str.replace(r"(\w+) _ (\w+)", r"\1 \2", regex=True)
+        .str.replace("new orleans harbor", "orleans harbor", regex=False)
+        .str.replace(
+            "probation & parcole - adult",
+            "adult probation",
+            regex=True,
+        )
+        .str.extract(
+            r"^(\w{1}? ?\w+ ?\w+? ?\w+) ?(deceased|full-time|reserve|retired|part-time)? ? "
+            r"?(\w{1,2}\/\w{1,2}\/\w{4}) ? ?(\w{1,2}\/\w{1,2}\/\w{4})? ?(termination|(\w+)? "
+            r"?(resignation)|(\w+)? ?(resigned)|retired)?"
+        )
+    )
+    df.loc[:, "agency_8"] = agency8[0]
+    df.loc[:, "employment_status_8"] = agency8[1]
+    df.loc[:, "hire_date_8"] = agency8[2]
+    df.loc[:, "left_date_8"] = agency8[3]
+    df.loc[:, "left_reason_8"] = agency8[4]
+    return df
```

Comment:
This looks like a lot of copy-pasted code. Can you extract this out to functions? Remember DRY principle.

Reply:
Thank you for the reminder!